### DISCUSSION TITLE
How to color svg?

### DISCUSSION THREAD

**Discussion by gyzerok** [emilk/egui#1668] [Q&A] [answered]
Hello!

I building an application which has lots of monochrome svg icons. These icons should be displayed with different colors in different cases.

On one hand I could copy this svgs and put different colors into them. However ideally I would like to be able to load them and specifying the color I would want to use. This is especially convenient since I store theme colors inside my code - when color changes I would need to update only single place.

Does it make sense at all? Would it be reasonable to add this feature?

**Comment by emilk** [comment, marked as answer]
Use `Image::tint`: https://docs.rs/egui/latest/egui/widgets/struct.Image.html#method.tint

**Reply by NiceneNerd** [reply]
Hmm, but how can you do this make light colors or white? When I load a `RetainedImage` from SVG data (in this case without color specified), it's black by default, and multiplying can't seem to do anything with it.

**Reply by xangelix** [reply]
For anyone with this same issue, "Tint" is not a misnomer here. Anything times 0 is 0, so if you have an SVG set to black, multiplying it by anything will still result in just black.
If you want perfectly accurate control over the color with tint, you'll want to set the base draw colors to 1, such that it perfectly adopts the tint.

This certainly doesn't solve every problem right now with configuring SVG draw in egui at a high level, but `fill="#ffffff"` inside your SVG file would probably fix your issue here.

**Comment by gyzerok** [comment]
Yeah, seems to work. Perfect, thanks!

As a side note - maybe worth giving `RetainedImage` a function which would return `image` to make code a bit simpler.

```rust
let muted_icon = cache.icons.muted_16();
let muted_icon =
    egui::Image::new(muted_icon.texture_id(ui.ctx()), muted_icon.size_vec2())
        .tint(Theme::icon_tertiary());
 ui.add(muted_icon);
```
